File: engine/transpile/src/test/resources/golden/SYK009/WS_確認メッセージ.py
"""SYK009 のデータ部レコード WS-確認メッセージ を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_確認メッセージ:
    """01 WS-確認メッセージ 総バイト長 40。"""

    LENGTH = 40

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(40)
        else:
            self._data = bytearray(data)
            if len(self._data) < 40:
                self._data.extend(b"\x00" * (40 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-確認メッセージ
    def get_WS_確認メッセージ(self):
        return decode_alphanumeric(self._data, 0, 40)
    def set_WS_確認メッセージ(self, value):
        encode_alphanumeric(self._data, 0, 40, value)

File: engine/transpile/src/test/resources/golden/SYK009/WS_確認メッセージ.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK009 のデータ部レコード WS-確認メッセージ を逐語対訳した自動生成コード。 */
public final class WS_確認メッセージ {

    /** 01 WS-確認メッセージ 総バイト長 40。 */
    public static final int LENGTH = 40;

    private final byte[] data;

    public WS_確認メッセージ() {
        this.data = new byte[40];
    }

    public WS_確認メッセージ(byte[] source) {
        this.data = Arrays.copyOf(source, 40);
    }

    public byte[] data() {
        return data;
    }

    // WS-確認メッセージ
    public String get_WS_確認メッセージ() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 40);
    }
    public void set_WS_確認メッセージ(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 40, value);
    }
}
